### PR TITLE
Add documentation about set_color_by_tex

### PR DESCRIPTION
**Pull request by Darylgolden**
## Motivation
`set_color_by_tex` can be confusing to use and not provide the result a user expects (only coloring the specific symbol or expression provided).

## Explanation for Changes
I've added documentation explaining what `set_color_by_tex` really does, and explained how to only color a specific symbol or expression.

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [ ] My new functions/classes either have a docstring or are private
- [ ] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

File: manim/mobject/svg/tex_mobject.py [hunk 1]
```diff
@@ -113,6 +113,38 @@ def construct(self):
             tex.set_color_by_tex('igsta', RED)
             self.add(tex)
 
+Note that :func:`~.set_color_by_tex`: colors the entire substring containing the Tex searched for,
+not just the specific symbol or Tex expression searched for. For example, consider the following example:
```
Comment (kolibril13) [inline manim/mobject/svg/tex_mobject.py:117]:
```suggestion
not just the specific symbol or Tex expression searched for. Consider the following example:
```

File: manim/mobject/svg/tex_mobject.py [hunk 2]
```diff
@@ -113,6 +113,38 @@ def construct(self):
...
+        def construct(self):
+            equation = MathTex(
+                "e^x = x^0 + x^1 + \\frac{1}{2} x^2 + \\frac{1}{6} x^3 + \\cdots + \\frac{1}{n!} x^n + \\cdots", substrings_to_isolate="x"
+                )
+            xs = equation.get_parts_by_tex("x")
+            xs.set_color(YELLOW)
+            self.add(equation)
+
+By setting `substring_to_isolate` to `x`, we split up the :class:`~.MathTex` into substrings automatically and in a way that isolates `x`
+into its own individual substrings. :func:`~.set_color_by_tex`: can then be used to achieve the desired result.
```
Comment (kolibril13) [inline manim/mobject/svg/tex_mobject.py:146]:
```suggestion
By setting ``substring_to_isolate`` to ``x``, we split up the :class:`~.MathTex` into substrings automatically and isolate ``x`` components into individual substrings. :func:`~.set_color_by_tex`: can then be used to achieve the desired result.
```

**Comment by kolibril13** [comment]
Thanks for the contribution.
As https://manimce--1269.org.readthedocs.build/en/1269/reference/manim.mobject.svg.tex_mobject.html is growing now bigger and bigger, I think it might be worth considering to add a tutorial about Text and shift the examples there for more visibility.

**Review by kolibril13 — APPROVED** [review]

File: manim/mobject/svg/tex_mobject.py [hunk 3]
```diff
@@ -113,6 +113,37 @@ def construct(self):
...
+    class IncorrectLaTeXSubstringColoring(Scene):
+        def construct(self):
+            equation = MathTex(
+                "e^x = x^0 + x^1 + \\frac{1}{2} x^2 + \\frac{1}{6} x^3 + \\cdots + \\frac{1}{n!} x^n + \\cdots", isolate="x"
+                )
+            xs = equation.get_parts_by_tex("x")
+            xs.set_color(YELLOW)
+            self.add(equation)
+
+colors the entire equation yellow, contrary to what may be expected. To color only `x` yellow, we have to do the following:
```
Comment (behackl) [inline manim/mobject/svg/tex_mobject.py:131]:
```suggestion
colors the entire equation yellow, contrary to what may be expected. To color only ``x`` yellow, we have to do the following:
```

File: manim/mobject/svg/tex_mobject.py [hunk 4]
```diff
@@ -113,6 +113,37 @@ def construct(self):
             tex.set_color_by_tex('igsta', RED)
             self.add(tex)
 
+Note that :func:`~.set_color_by_tex`: colors the entire substring containing the Tex searched for,
```
Comment (behackl) [inline manim/mobject/svg/tex_mobject.py:116]:
```suggestion
Note that :func:`~.set_color_by_tex` colors the entire substring containing the Tex searched for,
```

File: manim/mobject/svg/tex_mobject.py (hunk 3 above)
Comment (behackl) [reply manim/mobject/svg/tex_mobject.py:131]: Also, this doesn't read like a proper sentence together with the part before the example:

"Consider the following example: ... colors the entire equation yellow ..."

I'd rather go for something like

```
Consider the following example.

...

As you can see, with this approach the entire equation is colored yellow, contrary to what may me expected. ...
```

File: manim/mobject/svg/tex_mobject.py [hunk 5]
```diff
@@ -113,6 +113,37 @@ def construct(self):
...
+not just the specific symbol or Tex expression searched for. Consider the following example:
+
+.. manim:: IncorrectLaTeXSubstringColoring
+    :save_last_frame:
+
+    class IncorrectLaTeXSubstringColoring(Scene):
+        def construct(self):
+            equation = MathTex(
+                "e^x = x^0 + x^1 + \\frac{1}{2} x^2 + \\frac{1}{6} x^3 + \\cdots + \\frac{1}{n!} x^n + \\cdots", isolate="x"
```
Comment (behackl) [inline manim/mobject/svg/tex_mobject.py:125]: Where does the keyword `isolate` come from?

Comment (behackl) [reply manim/mobject/svg/tex_mobject.py:125]: Also, shouldn't this example actually use `set_color_by_tex`, as announced in the paragraph above it?

**Review by behackl — CHANGES_REQUESTED** [review]
Some of the proposed changes are not clear to me, please clarify.

File: manim/mobject/svg/tex_mobject.py (hunk 5 above)
Comment (Darylgolden) [reply manim/mobject/svg/tex_mobject.py]: Fixed.

File: manim/mobject/svg/tex_mobject.py [hunk 6]
```diff
@@ -113,6 +113,37 @@ def construct(self):
...
+    :save_last_frame:
+
+    class IncorrectLaTeXSubstringColoring(Scene):
+        def construct(self):
+            equation = MathTex(
+                "e^x = x^0 + x^1 + \\frac{1}{2} x^2 + \\frac{1}{6} x^3 + \\cdots + \\frac{1}{n!} x^n + \\cdots"
+                )
+            xs = equation.get_parts_by_tex("x")
+            xs.set_color_by_tex(YELLOW)
+            self.add(equation)
```
Comment (behackl) [inline manim/mobject/svg/tex_mobject.py:129]: I still feel that the incorrect example is too complicated. Why not simply: 

```suggestion
            equation.set_color_by_tex("x", YELLOW)
            self.add(equation)
```

(That's at least what I would initially expect to work.)

File: manim/mobject/svg/tex_mobject.py [hunk 7]
```diff
@@ -113,6 +113,37 @@ def construct(self):
...
+.. manim:: CorrectLaTeXSubstringColoring
+    :save_last_frame:
+
+    class CorrectLaTeXSubstringColoring(Scene):
+        def construct(self):
+            equation = MathTex(
+                "e^x = x^0 + x^1 + \\frac{1}{2} x^2 + \\frac{1}{6} x^3 + \\cdots + \\frac{1}{n!} x^n + \\cdots", substrings_to_isolate="x"
+                )
+            xs = equation.get_parts_by_tex("x")
+            xs.set_color(YELLOW)
```
Comment (behackl) [inline manim/mobject/svg/tex_mobject.py:142]: Same here.

```suggestion
            equation = MathTex(
                r"e^x = x^0 + x^1 + \frac{1}{2} x^2 + \frac{1}{6} x^3 + \cdots + \frac{1}{n!} x^n + \cdots",
                substrings_to_isolate="x"
            )
            equation.set_color_by_tex("x", YELLOW)
```

File: manim/mobject/svg/tex_mobject.py [hunk 8]
```diff
@@ -113,6 +113,37 @@ def construct(self):
...
+Note that :func:`~.set_color_by_tex` colors the entire substring containing the Tex searched for,
+not just the specific symbol or Tex expression searched for. Consider the following example:
+
+.. manim:: IncorrectLaTeXSubstringColoring
+    :save_last_frame:
+
+    class IncorrectLaTeXSubstringColoring(Scene):
+        def construct(self):
+            equation = MathTex(
+                "e^x = x^0 + x^1 + \\frac{1}{2} x^2 + \\frac{1}{6} x^3 + \\cdots + \\frac{1}{n!} x^n + \\cdots"
```
Comment (behackl) [inline manim/mobject/svg/tex_mobject.py:125]:
```suggestion
                r"e^x = x^0 + x^1 + \frac{1}{2} x^2 + \frac{1}{6} x^3 + \cdots + \frac{1}{n!} x^n + \cdots"
```

File: manim/mobject/svg/tex_mobject.py [hunk 9]
```diff
@@ -113,6 +113,37 @@ def construct(self):
...
+                )
```
Comment (behackl) [inline manim/mobject/svg/tex_mobject.py:126]:
```suggestion
            )
```

File: manim/mobject/svg/tex_mobject.py [hunk 10]
```diff
@@ -113,6 +113,37 @@ def construct(self):
...
+    class CorrectLaTeXSubstringColoring(Scene):
+        def construct(self):
+            equation = MathTex(
+                "e^x = x^0 + x^1 + \\frac{1}{2} x^2 + \\frac{1}{6} x^3 + \\cdots + \\frac{1}{n!} x^n + \\cdots", substrings_to_isolate="x"
+                )
+            xs = equation.get_parts_by_tex("x")
+            xs.set_color(YELLOW)
+            self.add(equation)
+
+By setting ``substring_to_isolate`` to ``x``, we split up the :class:`~.MathTex` into substrings automatically and isolate ``x`` components into individual substrings. :func:`~.set_color_by_tex`: can then be used to achieve the desired result.
```
Comment (behackl) [inline manim/mobject/svg/tex_mobject.py:145]: Minor rewording (I don't like sentences starting with functions), and removed a superfluous colon.

```suggestion
By setting ``substring_to_isolate`` to ``x``, we split up the :class:`~.MathTex` into substrings
automatically and isolate ``x`` components into individual substrings. Only then can :meth:`~.set_color_by_tex` be used to achieve the desired result.
```

**Review by behackl — CHANGES_REQUESTED** [review]
Another iteration (but this is it now, I promise). :-)

**Comment by jsonvillanueva** [comment]
I just realized this file has 188 lines of docstring for the module... why aren't these examples placed within their respective classes?

**Review by behackl — APPROVED** [review]
Thanks for your efforts!

**Comment by behackl** [comment]
> I just realized this file has 188 lines of docstring for the module... why aren't these examples placed within their respective classes?

Because there is some sort of storyline behind the documentation; in these cases I'd rather have all of it collected in one place, and the module docstring is semantically the correct place for it.

We do have to make sure that the documentation can be found, however -- I think that restructuring Manim's modules would help a lot with that already.

**Pull request merged**